src/decorator/decorators.ts

```diff
@@ -63,6 +63,22 @@ export function ValidateNested(validationOptions?: ValidationOptions) {
     };
 }
 
+/**
+ * Objects / object arrays marked with this decorator will also be validated.
+ */
+export function ValidateIf(condition: (object: any, value: any) => boolean, validationOptions?: ValidationOptions) {
```

Comment:
not sure if I like this name `ValidateIf`. Specifically I don't like `Validate` word here. Maybe we can come up with better name?